test/ibmq/runtime/test_runtime_integration.py

```diff
@@ -149,6 +149,11 @@ def test_upload_program(self):
         self.assertTrue(program)
         self.assertEqual(max_execution_time, program.max_execution_time)
 
+    def test_set_visibility(self):
+        """Test setting the visibility of a program."""
+        program_id = self._upload_program()
+        self.provider.runtime.program_set_visibility(program_id, 'private')
```

Comment:
Did you check if retrieving a program now tells you its visibility? If so we can verify that it's updated.